src/service/workspace_wrapper.py

```diff
@@ -47,16 +84,24 @@ def get_object_metadata(
             "ignoreErrors": 1,
             "includeMetadata": 1,
         })
-        for info, upa in zip(res['infos'], upas):
+        set_objs = []
+        std_objs = []
+        for info, path, upa in zip(res['infos'], res['paths'], upas):
             if not info:
                 raise errors.DataPermissionError(
                     f"The workspace service disallowed access to object {upa}")
-            if allowed_types and info[2].split('-')[0] not in allowed_types:
+            info[10][WORKSPACE_UPA_PATH] = ";".join(path)
+            type_ = self._get_type(info)
+            if allowed_set_types and type_ in allowed_set_types:
+                set_objs.append(info[10])
+            elif allowed_types and type_ not in allowed_types:
                 raise errors.IllegalParameterError(
                     f"Workspace object {upa} is type {info[2]}, which is not one of "
                     + f"the allowed types: {allowed_types}"
                 )
-        return [info[10] for info in res['infos']]
+            else:
+                std_objs.append(info[10])
```

Comment:
do you need if-elif-else.  Why not just do one if statement.  

`if allowed_types and type_ not in allowed_types: raise Error
  
std_objs.append(info[10])
`

Reply:
If I do
```python
            if allowed_set_types and type_ in allowed_set_types:
                set_objs.append(info[10])
            else allowed_types and type_ not in allowed_types:
                raise errors.IllegalParameterError(
                    f"Workspace object {upa} is type {info[2]}, which is not one of "
                    + f"the allowed types: {allowed_types}"
                )
            std_objs.append(info[10])
```
... which is what I think you're saying (?) then set objects will also get added to std objects

If I'm not understanding your suggestion LMK

Reply:
Oh nv. I misread `set_objs` vs `std_objs`.